modules/mesh-layers/src/simple-mesh-layer/simple-mesh-layer.js

```diff
@@ -265,12 +237,11 @@ export default class SimpleMeshLayer extends Layer {
     }
 
     if (src) {
-      getTexture(gl, src).then(texture => {
-        this.setState({texture});
-        if (this.state.model) {
-          this.state.model.setUniforms({sampler: this.state.texture, hasTexture: 1});
-        }
-      });
+      const texture = getTextureFromData(gl, src);
+      this.setState({texture});
+      if (this.state.model) {
```

Comment:
Unrelated: why do we have the `this.state.model` check? 

It looks like if `model` is not set here, the texture will never be set? 

Does it ever happen?

Add a comment?